keeper/query_server_vaults.go

```diff
@@ -66,3 +66,19 @@ func (k vaultsQueryServer) PositionsByProvider(ctx context.Context, req *vaults.
 		Positions: positions,
 	}, nil
 }
+
+func (k vaultsQueryServer) Stats(ctx context.Context, req *vaults.QueryStats) (*vaults.QueryStatsResponse, error) {
+	if req == nil {
+		return nil, types.ErrInvalidRequest
+	}
+
+	vaultsStats := make(map[string]*vaults.Stats_Vault)
+	_ = k.Keeper.VaultsStats.Walk(ctx, nil, func(key int32, value vaults.Stats_Vault) (stop bool, err error) {
```

Comment:
Should we maybe not ignore the error here? Thinking about if there is a case for that though 🤔